src/spoke/VaultSpoke.sol

```diff
@@ -0,0 +1,433 @@
+// SPDX-License-Identifier: UNLICENSED
+// Copyright (c) 2025 Aave Labs
+pragma solidity 0.8.28;
+
+import {ERC20Upgradeable} from 'src/dependencies/openzeppelin-upgradeable/ERC20Upgradeable.sol';
+import {SafeERC20, IERC20} from 'src/dependencies/openzeppelin/SafeERC20.sol';
+import {SignatureChecker, ECDSA} from 'src/dependencies/openzeppelin/SignatureChecker.sol';
+import {EIP712} from 'src/dependencies/solady/EIP712.sol';
+import {MathUtils} from 'src/libraries/math/MathUtils.sol';
+import {EIP712Hash, EIP712Types} from 'src/libraries/EIP712Hash.sol';
+import {NoncesKeyed} from 'src/utils/NoncesKeyed.sol';
+
+import {IERC4626, IERC20Metadata} from 'src/dependencies/openzeppelin/IERC4626.sol';
+import {IERC20Permit} from 'src/dependencies/openzeppelin/IERC20Permit.sol';
+import {IVaultSpoke} from 'src/spoke/interfaces/IVaultSpoke.sol';
+import {IHub} from 'src/hub/interfaces/IHub.sol';
+
+/// @title VaultSpoke
+/// @author Aave Labs
+/// @notice ERC4626 compliant vault for hub's listed asset position management.
+/// @dev Connects to one listed asset, only responsible for tokenizing positions.
+/// @dev Share price accounting is maintained solely on the Hub.
+abstract contract VaultSpoke is IVaultSpoke, ERC20Upgradeable, NoncesKeyed, EIP712 {
+  using SafeERC20 for IERC20;
+  using MathUtils for uint256;
+  using EIP712Hash for *;
+
+  IHub internal immutable HUB;
+  uint256 internal immutable ASSET_ID;
+  address internal immutable ASSET;
+  uint8 internal immutable DECIMALS;
+
+  /// @inheritdoc IVaultSpoke
+  uint40 public immutable MAX_ALLOWED_SPOKE_CAP;
+
+  /// @inheritdoc IVaultSpoke
+  uint192 public constant PERMIT_NONCE_KEY = 0;
+
+  constructor(address hub_, uint256 assetId_) {
+    HUB = IHub(hub_);
+    ASSET_ID = assetId_;
+    require(ASSET_ID < HUB.getAssetCount());
```

Comment:
```suggestion
    require(assetId_ < HUB.getAssetCount());
    HUB = IHub(hub_);
    ASSET_ID = assetId_;
```
we usually input check first

Reply:
Tho doing `IHub(hub_).getAssetCount())` in the require would allow to ensure the passed Hub contract follows the interface correctly